File: python/modules/auxiliary.py
# Importa todas as funções de operações
from modules.operacoes import somar, diminuir, multiplicar, dividir, expoenciar, radiciar, restoDivisao


# Função para ver se o número é válido
def validarNumero(frase: str) -> float:
    while True:
        try:
            return float(input(frase).replace(',', '.'))
        except ValueError:
            print('Este não é um número válido. Tente novamente.\n')


# Função para validar, ver se a operação é válida
def validarOperacao(frase: str) -> str:
    operacoes = ['+', '-', '*', '/', '**', '//', '%']
    while True:
        operacao = input(frase).strip()
        if operacao in operacoes:
            return operacao
        else:
            print('Esta não é uma operação válida. Tente novamente.\n')


# Função para calcular o resultado a partir da operação e de dois valores
def calcularResultado(operacao: str, a: float, b: float) -> float:
    match operacao:
        case '+': return somar(a, b)
        case '-': return diminuir(a, b)
        case '*': return multiplicar(a, b)
        case '/': return dividir(a, b)
        case '**': return expoenciar(a, b)
        case '//': return radiciar(a, b)
        case '%': return restoDivisao(a, b)

File: ruby/modules/auxiliary.rb
require_relative 'operacoes'

# Função p/ validar um número, com base num input
def validarNumero(frase)
  while true
    print frase
    entrada = gets.chomp.to_f

    if entrada != 0.0
      return entrada
    else
      puts "Este não é um número válido. Tente novamente.\n\n"
    end
  end
end


# Função p/ validar operação, caso for válida, com um input
def validarOperacao(frase)
  operacoes = ['+', '-', '*', '/', '**', '//', '%']
  while true
    print frase
    operacao = gets.chomp

    if operacoes.include?(operacao)
      return operacao
    else
      puts "Esta não é uma operação válida. Tente novamente.\n\n"
    end
  end
end


# Função p/ calcular resultado a partir da operação e dois números
def calcularResultado(operacao, a, b)
  case operacao
    when "+" 
      return somar(a, b)
    when "-"
      return diminuir(a, b)
    when "*"
      return multiplicar(a, b)
    when "/"
      return dividir(a, b)
    when "**"
      return expoenciar(a, b)
    when "//"
      return radiciar(a, b)
    when "%"
      return restoDivisao(a, b)
  end
end